WpfSampleDi/UserControlExample.xaml.cs

```diff
@@ -1,46 +1,12 @@
-﻿using System;
-using System.Windows;
-using System.Windows.Controls;
-using System.Windows.Navigation;
-using WpfSampleDi.Service;
+﻿namespace WpfSampleDi;
 
-namespace WpfSampleDi
+/// <summary>
+/// Interaction logic for UserControlExample.xaml
+/// </summary>
+public partial class UserControlExample
 {
-    /// <summary>
-    /// Interaction logic for UserControlExample.xaml
-    /// </summary>
-    public partial class UserControlExample : UserControl
+    public UserControlExample()
     {
-        private readonly ILogBase _logBase;
-
-        public UserControlExample()
-        {
-            InitializeComponent();
-
-            if (MainWindow.AppWindow?.LogBase != null)
-            {
-                _logBase = MainWindow.AppWindow.LogBase;
-            }
-        }
-
-        private void SetButton_Click(object sender, RoutedEventArgs e)

```

Comment:
Recommended way for WPF is to adhere to MVVM. That is:
1. You don't want to write click handlers directly in the View, instead you bind the button to a command in the ViewModel.
2. You don't want to access something like `Text` in a `TextBox` directly, instead you bind it to a string property in the ViewModel.